File: solutions/patterns/star-patterns/10-hollow-square/solution.java
import java.util.*;

class solution {
    public static void main(String args[]) {
        int n = 5;
        for (int i = 1; i <= n; i++) {
             // first column and last column
            if (i == 1 || i == n) {
                for (int j = 1; j <= n; j++) {
                    System.out.print("* ");
                }
            } 
            //hollow
             else  {
                for (int j = 1; j <= n; j++) {
                     // start and end
                    if (j == 1 || j == n) {
                        System.out.print("* ");
                    }
                    // hollow
                    else System.out.print("  ");
                }
            }
            System.out.println();
        }
    }
}

File: solutions/patterns/star-patterns/10-hollow-square/solution.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    cout << "Enter the N value: ";
    cin >> n;

    cout << "\nPrinting the pattern...\n\n";

    // Loop through each row
    for (int row = 1; row <= n; row++)
    {
        // Loop through each column
        for (int col = 1; col <= n; col++)
        {
            // Print stars for the first and last row
            if (row == 1 || row == n)
            {
                cout << "* ";
            }
            else
            {
                // For middle rows, print stars only at the borders
                if (col == 1 || col == n)
                    cout << "* ";
                else
                    cout << "  ";
            }
        }

        // Move to next line after each row
        cout << endl;
    }

    return 0;
}
